### ISSUE TITLE
Missleading calculations of language widgets

### ISSUE THREAD

**Issue by milotype** [WeblateOrg/weblate#4193]
**Describe the bug**
When switching between "all" languages and one specific language for the generation of widgets, the _percentages_ are recalculated correctly, but the _number of languages_ stays the same! 
Thus the **Big status badge** and the **Open Graph image** are missleading, when only one language is selected for promotion.

**To Reproduce**
Please see screenshots below, or check https://hosted.weblate.org/widgets/planner/?lang=hr

**Expected behavior**
When only one language is selected, only **Status badge** and the **Small status badge** should be shown.

**Screenshots**

ALL LAGUAGES
![WidgetCalculationAll](https://user-images.githubusercontent.com/43657314/87798129-e8ad5f00-c84b-11ea-8aa6-c77dcf9cce0d.png)

ONE LAGUAGE
![WidgetCalculationOne](https://user-images.githubusercontent.com/43657314/87798135-ec40e600-c84b-11ea-9bf4-db85086e234e.png)

**Comment by nijel** [comment]
What would you expect there in single language case? Should it be ommitted?

**Comment by milotype** [comment]
@nijel 
In single language case, I'd expect to see only `Status badge` and `Small status badge`.

I must admit, I do not quite understand the diverse "Components" calculations.
Please consider to merge this issue with #4192.

**Comment by milotype** [comment]
Sorry, I forgot to explain the "Components". See screenshot

![Components](https://user-images.githubusercontent.com/43657314/87863729-a0329600-c95e-11ea-8c9b-a38fdab87776.png)

**Comment by nijel** [comment]
> I must admit, I do not quite understand the diverse "Components" calculations.

This gives stats per component.

**Comment by milotype** [comment]
Yes, I know. Sorry, I wanted to say, that I don't know how they would impact on the calculations.
